### ISSUE TITLE
Error when upgrading a Windows agent via WPK with a manager v4.4

### ISSUE THREAD

**Issue by MarcelKemp** [wazuh/wazuh#15424]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.4 | WPK upgrade | Manager/Agent | Sources | Ubuntu 20/Windows 2022 |

## Description

During the manual testing of the WPK upgrade for `v4.3.10` (https://github.com/wazuh/wazuh/issues/15388#issuecomment-1317013976), a bug has been found, which does not allow upgrading the Windows agent if a `v4.4` manager is used. This happens when trying to upgrade from v4.4 manager to Windows agent (tested with `v4.3.8`, `v4.3.9` and `v4.3.10` to `v4.3.10`).

- **Manager**: v4.4.0 - Branch used for the build: [4.4](https://github.com/wazuh/wazuh/tree/4.4) (with the last commit: https://github.com/wazuh/wazuh/commit/b99a2dd76bfe7b88c3ef8127cf5521f5fad3833a)
- **Agent**: Packages [v4.3.8](https://packages.wazuh.com/4.x/windows/wazuh-agent-4.3.8-1.msi), [v4.3.9](https://packages.wazuh.com/4.x/windows/wazuh-agent-4.3.9-1.msi) or [v4.3.10](https://packages.wazuh.com/4.x/windows/wazuh-agent-4.3.10-1.msi)
- **WPK**: Current [WPK v4.3.10](https://packages.wazuh.com/4.x/wpk/windows/wazuh_agent_v4.3.10_windows.wpk)

## Steps to reproduce

I have tested it using the following steps, both through the `API` and the `agent_upgrade` tool.

<details><summary>Common steps</summary>

<br>

1. Install manager, in this case I have done it by sources:
```
curl -sL https://github.com/wazuh/wazuh/tarball/4.4 | tar zx
cd wazuh-wazuh-b99a2dd/
./install.sh
```
2. Install agent, downloading the official package [v4.3.9](https://packages.wazuh.com/4.x/windows/wazuh-agent-4.3.9-1.msi), and connect it to the manager.

3. Download the [Windows WPK (v4.3.10)](https://packages.wazuh.com/4.x/wpk/windows/wazuh_agent_v4.3.10_windows.wpk) in the manager and move it to the `/var/ossec` folder:
```
wget https://packages.wazuh.com/4.x/wpk/windows/wazuh_agent_v4.3.10_windows.wpk
mv wazuh_agent_v4.3.10_windows.wpk /var/ossec/
```
</details>

<details><summary>Validate the upgrade using the `agent_upgrade` tool 🔴</summary>

<br>

4. In the manager, settle in the `/var/ossec` folder and run the `agent_upgrade`:
```
cd /var/ossec/
/var/ossec/bin/agent_upgrade -a 001 -f wazuh_agent_v4.3.10_windows.wpk
```

And the output it shows is as follows:
```
Upgrading...

Failed upgrades:
	Agent 001 status: Upgrade procedure exited with error code
```

> **Conclusion**: The upgrade process failed.

</details>

<details><summary>Validate the upgrade using the `API` 🔴</summary>

<br>

4. In the manager, get the TOKEN and run the upgrade command:
```
TOKEN=$(curl -u wazuh:wazuh -k -X GET "https://localhost:55000/security/user/authenticate?raw=true")
curl -k -X PUT "https://localhost:55000/agents/upgrade_custom?agents_list=001&file_path=/var/ossec/wazuh_agent_v4.3.10_windows.wpk&pretty=true" -H  "Authorization: Bearer $TOKEN"
```
```
{
   "data": {
      "affected_items": [
         {
            "agent": "001",
            "task_id": 2
         }
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All upgrade tasks were created",
   "error": 0
}
```
5. Check the upgrade status (where we can see the **_Error_** status):
```
curl -k -X GET "https://localhost:55000/agents/upgrade_result?agents_list=001&pretty=true" -H  "Authorization: Bearer $TOKEN"
```
```
{
   "data": {
      "affected_items": [
         {
            "message": "Success",
            "agent": "001",
            "task_id": 2,
            "node": "node01",
            "module": "upgrade_module",
            "command": "upgrade_custom",
            "status": "Error",
            "error_msg": "Upgrade procedure exited with error code",
            "create_time": "2022-11-17T12:56:29Z",
            "update_time": "2022-11-17T12:58:10Z"
         }
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All upgrade tasks were returned",
   "error": 0
}
```

6. Make sure that the version has not changed:
```
curl -k -X GET "https://localhost:55000/agents?agents_list=001&pretty=true&select=version" -H  "Authorization: Bearer $TOKEN"
```
```
{
   "data": {
      "affected_items": [
         {
            "id": "001",
            "version": "Wazuh v4.3.9"
         }
      ],
      "total_affected_items": 1,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All selected agents information was returned",
   "error": 0
}
```

> **Conclusion**: The upgrade process failed.

</details>

### Extra info

- [Manager's log](https://github.com/wazuh/wazuh/files/10032816/manager.log)
  - Failures can be observed as per `update` and `remoted` per agent key already in use.
- [Agent's log](https://github.com/wazuh/wazuh/files/10032861/agent.log)
- [Upgrade.log](https://github.com/wazuh/wazuh/files/10032865/upgrade.log)

- Validate that the error does not appear in the production environment (🟢), can be checked with a v4.3.10 manager in release testing: https://github.com/wazuh/wazuh/issues/15388#issuecomment-1317013976

## Expected results

The upgrade process must be completed successfully.

**Comment by MarcelKemp** [comment]
### Tested cases
After a thorough researching of the case, we have found the following cases:
- Successfully:
  - Running a WPK upgrade with a `v4.3.10` manager to upgrade an agent to `v4.3.10` or lower (issue with testing: https://github.com/wazuh/wazuh/issues/15388#issuecomment-1317013976).
  - Running a WPK upgrade with a `v4.4` manager to upgrade an agent to `v4.4` (issue with testing: https://github.com/wazuh/wazuh/pull/15322)

- Failed:
  - Running a WPK upgrade with a `v4.4` manager to upgrade an agent to `v4.3.10` or lower (reported in the [description](https://github.com/wazuh/wazuh/issues/15424#issue-1453534566)).

### Workaround
Knowing the failed case in detail, a test has been made to generate a custom Windows WPK for `v4.3.10`, where use is made of the Windows scripts containing the following fix:
- https://github.com/wazuh/wazuh/pull/15322

And the tests have been successful, being able to upgrade with a `v4.4` manager to a `v4.3.10` agent.


### Options to determine
Having this information, it would only remain to determine an action on the issue, having available among some options:
- Apply a workaround where we generate all the WPK packages of `v4.3.10` with the fixed scripts and comment it in the custom WPK generation guide.
- Apply fix [#15322](https://github.com/wazuh/wazuh/pull/15322) to the production branches.
- Don't do anything about it, but inform the users that it would not be possible to upgrade with a `v4.4` manager an agent to `v4.3.10` or lower, only being possible to upgrade the agent by WPK to `v4.4`.

**Comment by MarcelKemp** [comment]
In the end, it has been decided not to take any action on the issue, so it should be noted that from a `v4.4` manager it will not be possible to upgrade a Windows agent via WPK to a `v4.3.10` or lower version, except for applying the [workaround](https://github.com/wazuh/wazuh/issues/15424#issuecomment-1320076534) mentioned above.
